tests/integration/panic.rs

```diff
@@ -0,0 +1,341 @@
+use chalk_integration::interner::{ChalkIr, RawId};
+use chalk_ir::*;
+use chalk_rust_ir::*;
+use chalk_solve::{RustIrDatabase, SolverChoice};
+use std::sync::Arc;
+
+#[derive(Debug)]
+enum PanickingMethod {
+    NoPanic,
+    CustomClauses,
+    TraitDatum,
+    ImplDatum,
+    ImplsForTrait,
+    ProgramClausesForEnv,
+    Interner,
+}
+
+impl Default for PanickingMethod {
+    fn default() -> Self {
+        Self::NoPanic
+    }
+}
+
+#[derive(Debug, Default)]
+struct MockDatabase {
+    panicking_method: PanickingMethod,
+}
+
+/// This DB is representint lowered program:
+///
+/// struct Foo { }
+/// trait Bar { }
+/// impl Bar for Foo { }
+// FIXME: Check if items returned from functions of this struct can be simplified
```

Comment:
Apparently this didn't get saved before I submitted the review:

I would move this to the top of the file above `PanickingMethod` and save "FIXME: some of these are probably redundant, so we should figure out which panic in the same place in `chalk-engine`"